### PR TITLE
feat(core): implement parts of workspace upgrade design

### PR DESCRIPTION
**Pull request by joooye34**
https://github.com/toeverything/AFFiNE/assets/9527419/704f4615-43f5-4cca-9428-9b0bf642461b

https://www.figma.com/file/rfH3P9Wbz4IEp2Ai18O8dV/%F0%9F%96%8C%EF%B8%8F-2023.Late?node-id=8542%3A13129&mode=dev

### CODE REVIEW

**Review by pengx17 — APPROVED** [review]

File: packages/frontend/core/src/components/workspace-upgrade/upgrade-hooks.ts [hunk 1]
```diff
@@ -33,46 +36,50 @@ export const MigrationFallback = function MigrationFallback() {
...
+          remoteProvider.datasource
+        );
+      }
+    })()
+      .then(() => setState('done'))
+      .catch((e: any) => {
+        console.error(e);
+        setError(e);
+        setState('error');
+      });
```
[the remark this answers is not in the document]
Comment (joooye34) [reply packages/frontend/core/src/components/workspace-upgrade/upgrade-hooks.ts:76]: It is not in one executing scope. This is a forced way to run async function in hooks callback.

[the remark this answers is not in the document]
Comment (joooye34) [reply packages/frontend/core/src/components/workspace-upgrade/upgrade-hooks.ts:76]: The async callback should not be defined, the async logic should with a then/catch.

[the remark this answers is not in the document]
Comment (joooye34) [reply packages/frontend/core/src/components/workspace-upgrade/upgrade-hooks.ts:76]: but you are right, then function should be improved.

[the remark this answers is not in the document]
Comment (joooye34) [reply packages/frontend/core/src/components/workspace-upgrade/upgrade-hooks.ts]: Then function have been removed, and catch should be keepped.

**Review by pengx17 — APPROVED** [review]

**Pull request merged**